controllers/propagator/propagation.go

```diff
@@ -945,6 +993,30 @@ func (r *PolicyReconciler) processTemplates(
 		}
 
 		policyT.ObjectDefinition.Raw = resolveddata
+
+		// Set initialization vector annotation on the ObjectDefinition for the controller's use
+		if usesEncryption {
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+
+			jsonErr := json.Unmarshal(resolveddata, policyTObjectUnstructured)
+			if jsonErr != nil {
+				// it shouldn't get here but if it did just log a msg
+				// it's all right, a generic msg will be used on the managedcluster
+				log.Error(jsonErr, "Error unmarshalling the object definition to JSON")
```

Comment:
Like I said previously, you'll want to return this error since we don't want to continue processing the templates if the annotation can't get added.

Reply:
I returned the error below this. Should it be returned both places?

Reply:
Silly question. I'll update it.